py/_path/local.py

```diff
@@ -820,16 +820,19 @@ def parse_num(path):
                 except ValueError:
                     pass
 
-        exclusive_flag = 'x' if not sys.platform.startswith('win') else ''
-
         def create_lockfile(path):
             """ exclusively create lockfile. Throws when failed """
             mypid = os.getpid()
             lockfile = path.join('.lock')
             if hasattr(lockfile, 'mksymlinkto'):
                 lockfile.mksymlinkto(str(mypid))
             else:
-                lockfile.write(str(mypid), 'w' + exclusive_flag)
+                try:
+                    fd = os.open(str(lockfile), os.O_WRONLY | os.O_CREAT | os.O_EXCL, 0o644)
```

Comment:
use `py.error.checked_call` here

Reply:
main reason is to account for all the other errors that could **unexpectedly** happen